packages/js-connect-with-portabl/src/index.ts

```diff
@@ -1,46 +1,89 @@
 import jwtDecode from 'jwt-decode';
+import { verify as verifyJWT } from 'jsonwebtoken';
 
 import { Transaction, TransactionManager } from './lib/transaction-manager';
 import { SessionStorage } from './lib/storage';
 import { createRandomString } from './lib/random-string';
 import {
+  ERROR_PREFIX,
   CONSOLE_LOG_PREFIX,
   ID_TOKEN_LOCAL_STORAGE_KEY,
 } from './lib/constants';
 
 export interface IConnectClientOptions {
-  accountId: string;
-  connectDomain: string;
-  walletDomain: string;
+  readonly accountId: string;
+  readonly connectDomain: string;
+  readonly walletDomain: string;
 }
 
-export type GetAccessTokenResponse = {
-  access_token: string;
-};
+export interface ITokenEndpointResponse {
+  readonly access_token: string;
+}
+
+interface IChallengeOptions {
+  readonly challenge: string;
+  readonly domain: string;
+}
 
-type SingleProofType = { challenge: string; domain: string };
+interface ILdProof extends IChallengeOptions {}
 
-type ProofType = SingleProofType | Array<SingleProofType>;
+type LdProofType = ILdProof | Array<ILdProof>;
 
-interface IVPToken {
-  proof: ProofType;
+interface IVpToken {
+  readonly proof: LdProofType;
 }
 
-interface IDecodedIdToken {
-  exp: number;
-  nonce: string;
+interface IIdTokenPayload {
+  readonly iat: number;
+  readonly exp: number;
+  readonly iss: string;
+  readonly sub: string;
+  readonly aud: string;
+  readonly nonce: string;
 }
 
-type GetResponseDataResponse = {
-  id_token: string;
-  vp_token: IVPToken;
-};
+interface IGetAuthResponseDto {
+  readonly id_token: string;
+  readonly vp_token: IVpToken;
+}
+
+interface ICreateTransactionResponse {
+  readonly authRequestUri: string;
+  readonly transactionId: string;
+  readonly nonce: string;
+  readonly state: string;
+}
 
-type CreateTransactionResponse = {
-  authRequestUri: string;
-  transactionId: string;
-  nonce: string;
-  state: string;
+enum UserStatusEnum {
+  Registered = 'Registered', // Q: shall we consider changing it to "Active"?
+  Suspended = 'Suspended',
+  InReview = 'InReview',
```

Comment:
@bao-portabl good point
@benox3 and I thought of being more concrete by using `InReview` as essentially business is expected to execute a review of submitted application.
Also, we could think of having a more generic `Pending` status and a separate property describing more granular status behind `Pending` in case various options could be applicable like `InReview`, etc